src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_evp.c

```diff
@@ -93,6 +93,30 @@ int32_t CryptoNative_EvpDigestCurrent(const EVP_MD_CTX* ctx, uint8_t* md, uint32
     return 0;
 }
 
+int32_t CryptoNative_EvpDigestOneShot(const EVP_MD* type, const void* source, int32_t sourceSize, uint8_t* md, uint32_t* mdSize)
+{
+    if (type == NULL || sourceSize < 0 || md == NULL || mdSize == NULL)
+        return 0;
+
+    EVP_MD_CTX* ctx = CryptoNative_EvpMdCtxCreate(type);
```

Comment:
"Can't this be written in terms of the simple one-shot hash routines?"

\- "Well, they're marked as deprecated in OpenSSL 3.0, with this being the recommended pattern."

"Oh."